src/m365/entra/commands/enterpriseapp/enterpriseapp-remove.spec.ts

```diff
@@ -0,0 +1,395 @@
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if the appObjectId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { appObjectId: '123' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if both appId and appDisplayName are specified', async () => {
+    sinon.stub(cli, 'getSettingWithDefaultValue').callsFake((settingName, defaultValue) => {
+      if (settingName === settingsNames.prompt) {
+        return false;
+      }
+
+      return defaultValue;
+    });
+
+    const actual = await command.validate({ options: { appId: '123', appDisplayName: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if appObjectId and appDisplayName are specified', async () => {
+    sinon.stub(cli, 'getSettingWithDefaultValue').callsFake((settingName, defaultValue) => {
+      if (settingName === settingsNames.prompt) {
+        return false;
+      }
+
+      return defaultValue;
+    });
+
+    const actual = await command.validate({ options: { appDisplayName: 'abc', appObjectId: '123' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('supports specifying appId', () => {
+    const options = command.options;
+    let containsOption = false;
+    options.forEach(o => {
+      if (o.option.indexOf('--appId') > -1) {
+        containsOption = true;
+      }
+    });
+    assert(containsOption);
+  });
+
+  it('supports specifying appDisplayName', () => {
+    const options = command.options;
+    let containsOption = false;
+    options.forEach(o => {
+      if (o.option.indexOf('--appDisplayName') > -1) {
+        containsOption = true;
+      }
+    });
+    assert(containsOption);
+  });
+
+  it('prompts before removing the enterprise application when force option not passed', async () => {
+    await command.action(logger, { options: { appId: '65415bb1-9267-4313-bbf5-ae259732ee12' } });
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the enterprise application when prompt not confirmed', async () => {
+    sinonUtil.restore(cli.promptForConfirmation);
+    sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    const deleteCallsSpy: sinon.SinonStub = deleteRequestStub();
+    await command.action(logger, { options: { appId: '65415bb1-9267-4313-bbf5-ae259732ee12' } });
+    assert(deleteCallsSpy.notCalled);
+  });
+
+  it('deletes the specified enterprise application using its display name', async () => {
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/servicePrincipals?$filter=displayName eq 'foo'`) {
+        return spAppInfo;
+      }
+
+      throw 'Invalid request';
+    });
+
+    const deleteCallsSpy: sinon.SinonStub = deleteRequestStub();
+    await command.action(logger, { options: { debug: true, appDisplayName: 'foo', force: true } });
```

Comment:
Let's use `verbose: true` instead of `debug: true` in all tests where this is needed.